File: Arrays/LECANDY.py
T = int(input())

for i in range(T):
    temp = input().split(" ")
    N, C = int(temp[0]), int(temp[1])
    candy = [int(x) for x in input().split(" ")]

    if sum(candy) <= C:
        print("YES")
    else:
        print("NO")

File: Arrays/LECANDY.cpp
#include <iostream>
using namespace std;

int main(){
    int T, N, C, temp, sum;
    cin >> T;

    for (int i = 0; i < T; i++){
        cin >> N >> C;
        sum = 0;

        for (int j = 0; j < N; j++){
            cin >> temp;
            sum += temp;
        }

        (sum <= C) ? cout << "Yes\n" : cout << "No\n";
    }
    
    return 0;
}
